capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -12,19 +12,24 @@ import { HttpClientModule } from '@angular/common/http';
   styleUrls: ['./wiki-media.component.scss']
 })
 
-export class WikiMediaComponent implements OnInit, OnDestroy {
+export class WikiMediaComponent implements OnChanges, OnInit {
   history: string;
   title: string;
   error: string;
   destroy$ = new Subject<void>();
   loading = true;
   body: string;
   urls = new Map();
+  @Input() cityName!: string;
 
   constructor(private wikiService: WikiResultsService) { }
 
   ngOnInit(): void {
-    this.getResults('New York City, NY');
+    this.getResults(this.cityName);
+  }
+
+  ngOnChanges(): void {
+    this.getResults(this.cityName);
```

Comment:
I recommend restricting this action to changes of the cityName. This can be done using the simple changes interface of ngOnChanges.

Reply:
Isn't it already using ngOnChanges?

Reply:
Sorry, I'm a little confused...

Reply:
Sorry, let me explain better. Currently, getResults is called when ANY change occurs. I think this should be changed so that it only is called when "cityName" is changed. One way to do this is to use ngOnChanges' "SimpleChanges" interface. See https://angular.io/api/core/SimpleChanges and https://dev.to/nickraphael/ngonchanges-best-practice-always-use-simplechanges-always-1feg